appveyor/numpy.patch

```diff
@@ -0,0 +1,831 @@
+         self.fcompiler = None
+         self.inplace = 0
+         self.parallel = None
++        self.dlls = [os.environ['OPENBLAS_DLL']]
+ 
+     def finalize_options(self):
+         if self.parallel:
+@@ -75,7 +79,8 @@ class build_clib(old_build_clib):
+ 
+         for (lib_name, build_info) in self.libraries:
+             l = build_info.get('language', None)
+-            if l and l not in languages: languages.append(l)
++            if l and l not in languages:
++                languages.append(l)
+ 
+         from distutils.ccompiler import new_compiler
+         self.compiler = new_compiler(compiler=self.compiler,
+@@ -94,11 +99,11 @@ class build_clib(old_build_clib):
+         if self.have_f_sources():
+             from numpy.distutils.fcompiler import new_fcompiler
+             self._f_compiler = new_fcompiler(compiler=self.fcompiler,
+-                                               verbose=self.verbose,
+-                                               dry_run=self.dry_run,
+-                                               force=self.force,
+-                                               requiref90='f90' in languages,
+-                                               c_compiler=self.compiler)
++                                             verbose=self.verbose,
++                                             dry_run=self.dry_run,
++                                             force=self.force,
++                                             requiref90='f90' in languages,
++                                             c_compiler=self.compiler)
+             if self._f_compiler is not None:
+                 self._f_compiler.customize(self.distribution)
+ 
+@@ -114,10 +119,10 @@ class build_clib(old_build_clib):
+         self.build_libraries(self.libraries)
+ 
+         if self.inplace:
+-            for l in  self.distribution.installed_libraries:
++            for l in self.distribution.installed_libraries:
+                 libname = self.compiler.library_filename(l.name)
+                 source = os.path.join(self.build_clib, libname)
+-                target =  os.path.join(l.target_dir, libname)
++                target = os.path.join(l.target_dir, libname)
+                 self.mkpath(l.target_dir)
+                 shutil.copy(source, target)
+ 
+@@ -129,8 +134,37 @@ class build_clib(old_build_clib):
+         return filenames
+ 
+     def build_libraries(self, libraries):
++        library_names = [
++            'mach',
++            'dfftpack',
++            'fftpack',
++            'sc_mach',
++            'sc_amos',
++            'sc_cdf',
++            'sc_specfun',
++            'statlib',
++            'quadpack',
++            'dop',
++            'fitpack',
++            'fwrappers',
++            'odrpack',
++            'minpack',
++            'arpack_scipy',
++            'odepack',
++            'rootfind',
++            'superlu_src',
++            'sc_c_misc',
++            'sc_cephes',
++        ]
```

Comment:
Issue #1: the build_clib is reordered so that fortran libs can be linked against their dependencies.